calour/training.py

```diff
@@ -121,24 +126,202 @@ def split_train_test(exp: Experiment, field, test_size, train_size=None, stratif
     return train_X, test_X, train_y, test_y
 
 
-@Experiment._record_sig
-def collect_cv_prediction(exp: Experiment, field, estimator, cv, scoring, inplace=False):
-    '''Do the CV
+def classify(exp: Experiment, field, estimator, cv=RepeatedStratifiedKFold(3, 1),
+             predict='predict_proba', param_grid=None):
+    '''Evaluate classification during cross validation.
+
+    Parameters
+    ----------
+    field : str
+        column name in the sample metadata, which contains the classes we want to predict.
+    estimator : object
+        scikit-learn estimator. e.g. :class:`sklearn.ensemble.RandomForestClassifer`
+    cv : int, cross-validation generator or an iterable
+        similar to the `cv` parameter in :class:`sklearn.model_selection.GridSearchCV`
+    predict : {'predict', 'predict_proba'}
+        the method to predict the validation sets
+    param_grid : dict of string to sequence, or sequence of such
```

Comment:
i don't understand what "dict of string to sequence" means... :(

Reply:
i think `string to sequence` means the dict map of a string to a sequence. copied from sklearn doc. I prefer not changing it.